mobile/src/viewModels/HomeViewModel.ts

```diff
@@ -168,19 +178,45 @@ export default class HomeViewModel {
         ];
     }
 
+    private submitPendingShortIfTimeForFull() {
+        if (AppController.Instance.User.localSettings?.current?.qol?.pendingShortQol && this.isTimeForFullQol()) {
+            this._settings.updateQolSettings({ pendingShortQol: false }, 'pendingShortQol');
+        }
+    }
+
     // returns true if it has been 28 calendar days since last Full QoL
+    // return true if there is a pending Full QoL
     private isTimeForFullQol(): boolean {
         const lastFullQol: Date = new Date(AppController.Instance.User.localSettings?.current?.qol?.lastFullQol);
-        let nextFullQol: Date = lastFullQol;
-        nextFullQol.setDate(nextFullQol.getDate() + 28);
+        const nextFullQol = new Date(lastFullQol.getDate() + 28);
+        const today: Date = new Date();
+
+        if (equalDateByDay(nextFullQol, today)) {
+            this._settings.updateQolSettings({ lastFullQol: Date() }, 'lastFullQol');
+            this._settings.updateQolSettings({ pendingFullQol: true }, 'pendingFullQol');
+            return true;
+        } else return AppController.Instance.User.localSettings?.current?.qol?.pendingFullQol;
+    }
+
+    // returns true if it has been 7 calendar days since last Short QoL
+    // return true if there is a pending Short QoL
+    private isTimeForShortQol(): boolean {
+        const lastShortQol: Date = new Date(AppController.Instance.User.localSettings?.current?.qol?.lastShortQol);
+        const nextShortQol = new Date(lastShortQol.getDate() + 7);
         const today: Date = new Date();
-        if (nextFullQol.getDay() === today.getDay() && nextFullQol.getMonth() === today.getMonth()
-        && nextFullQol.getFullYear() === today.getFullYear()) {
-            this._settings.updateLastFullQol({ lastFullQol: Date() });
-            this._settings.updatePendingFullQol({ pendingFullQol: true });
+
+        if (equalDateByDay(nextShortQol, today)) {
+            this._settings.updateQolSettings({ lastShortQol: Date() }, 'lastShortQol');
+            this._settings.updateQolSettings({ pendingShortQol: true }, 'pendingShortQol');
             return true;
-        } else if (AppController.Instance.User.localSettings?.current?.qol?.pendingFullQol) { return true; }
-        return false;
+        } else return AppController.Instance.User.localSettings?.current?.qol?.pendingShortQol;
+    }
+
+    private hasCompletedDailyCheckInToday(): boolean {
+        const lastDailyCheckIn: Date = new Date(AppController.Instance.User.localSettings?.current?.lastDailyCheckIn);
+        const today: Date = new Date();
+
+        return (lastDailyCheckIn.getDay() === today.getDay() && lastDailyCheckIn.getMonth() === today.getMonth() && lastDailyCheckIn.getFullYear() === today.getFullYear())
```

Comment:
to improve readability I'd recommend putting each && case on a new line. Can use prettier extension to improve formatting?